src/compiler/checker.ts

```diff
@@ -936,6 +937,34 @@ namespace ts {
             return false;
         }
 
+
+        function checkAndReportErrorForExtendingInterface(errorLocation: Node): boolean {
+            const container = getContainingClass(errorLocation);
+            const heritageClause = <HeritageClause>getAncestor(errorLocation, SyntaxKind.HeritageClause);
+            if (!container || !heritageClause || heritageClause.token !== SyntaxKind.ExtendsKeyword) {
+                return false;
+            }
+            if (errorLocation.kind === SyntaxKind.Identifier) {
+                const name = (<Identifier>errorLocation).text;
+                const interfaceOrModule = resolveName(

```

Comment:
Sorry wrong recommendation earlier. you probably should use `resolveEntityName` it handles both identifiers and propeortyAccess.